### DISCUSSION TITLE
Deleting Canvas Collections-Editing page doesn't unlock Editing

### DISCUSSION THREAD

**Discussion by omachado-id** [djplaner/canvas-collections#293] [Q&A] [answered]
Hi,

Just reporting another problem with which I need help saving myself from myself...

I was putting the final touches on a soon-to-be live course in which I've used Canvas Collections.  All has gone perfectly fine with the tool until this afternoon.  Then I've had this warning dialog pop up when I tried to do a bit more editing: 
![image](https://github.com/djplaner/canvas-collections/assets/143226390/6f42172d-174b-44ab-a191-68281aad0bc8)

Reading up on the Help literature, the suggestion to delete the Canvas Collections-Editing page has not unlocked the tool for me.  I've tried several times (each time I click on "Edit On" button, a new version of the Collections-Editing page is generated, but deleting all of them fails to reset the tool.

I am the only person who uses CC here and certainly the only person editing this course, so two people editing the same course is not the problem.  I've logged completely off the browser (latest version of Firefox), cleared its cache and relogged in to the course again, to no avail.  I've disabled and re-enabled CC in Tampermonkey.  I'm on CC version 1.2.2, which according to Tampermonkey, was updated about 30-ish hours ago (my time). Yet I continue to get the pictured error alert. 

Can you recommend some next steps that I can take to reset CC to a working state?

**Comment by djplaner** [comment]
Mmmm, a similar issue has arisen at my institution. Which is using a slightly different code base.  Just tested on my own and the same problem exists.  This is suggestive of a change in Canvas.  I will explore more and let you know.

**Reply by djplaner** [reply]
Just confirming that the problem appears to be a change in behaviour from Canvas when it comes to Canvas pages.  This change in turn breaks how Collections implements it edit lock (using a page).

The change from Canvas is that if you create any page (e.g. with a title like `Canvas Collections - Editing`), in Canvas it will initially have the name `canvas-collections-editing`.   Historically, if you then deleted it and then re-created a page with the same title, it would end up having the same name.

But now,  that re-created page will have the name `canvas-collections-editing-2`. Or -3, -4 depending on how many times you try.

We're digging into this. Trying to find out more from Instructure about the rationale for the change and also thinking about ways to work around it.

**Reply by djplaner** [reply]
Appears that the 16 September update to Canvas introduced a [change to pages](https://community.canvaslms.com/t5/Canvas-Releases/Canvas-Release-Notes-2023-09-16/ta-p/577167).  It's not directly related to this problem, but the timing is suspicious and suggestive of a causal link.   My current institution's technology folk are reaching out to Instructure for clarification.

If this is a long term intended change, Collections will need updating.   Though until then, we won't be able to make changes.

**Comment by omachado-id** [comment]
I understand completely, and thanks for investigating this!  It appears that once again, an seemingly innocuous change by Canvas: "9/16: In Pages, when editing a page title, all links to that page automatically update in a course," can have downstream effects that screw up everyone's day.  (This isn't the first time their changes have done this sort of thing, IME).  The cold comfort good news is that your code worked fine until Instructure's changes created the issue.

In any event, I'll await your findings and updates, David.  Good luck!

**Reply by djplaner** [reply, marked as answer]
We have a solution.  

The new version of Canvas Collections (v1.2.3) has been modified to handle this change in behaviour from Canvas.  In theory the update should come through automatically.  However, it may be quicker [to reinstall](https://djplaner.github.io/canvas-collections/getting-started/install/how-to-install/)

Further testing will be done to rule out any further issues, but it is looking good.  Please do share if you have any further related issues.

**Reply by omachado-id** [reply]
That's terrific news!

I've updated to v. 1.2.3 (no issues with the update).  Initial testing with the minor edits that I'd intended to make before the locking issue occurred went smoothly.  I was able to open up Editing, make the edits, save, turn off, then turn on editing several times without any lockups.  So my initial impression is that you've solved the issue.  I can now release the course for my faculty's use.  Thanks again for your efforts with this issue.

Further testing: Since I'm basically done with the course in question, I'll have to do more verification in another course in which I'm already planning to use CC to revise its look.  I'll start in on that--hopefully--today.  If anything untoward occurs, I'll report back.
